content/blog/scheduling-in-react/index.md

```diff
@@ -0,0 +1,243 @@
+---
+title: "Scheduling in React"
+date: "2019-03-15T12:00:00.000Z"
+---
+
+In modern applications, user interfaces often have to juggle multiple tasks at the same time: A search component needs to respond to user input while providing auto completion results; An interactive dashboard needs to update the charts while loading data from the server and sending analytics data to your backend.
+
+All these parallel steps can lead to janky interfaces and unhappy users, so let’s learn how we can fix this.
+
+## Scheduling in User Interfaces
+
+Our users expect immediate feedback. Whether they are clicking on a button to open a modal or adding text to an input field, they expect some kind of immediate confirmation. The button should show a modal and the input field should display the key that was typed.
+
+To visualize what happens when this is not the case, let’s take a look at the demo application that Dan Abramov presented at his talk, [Beyond React 16](https://reactjs.org/blog/2018/03/01/sneak-peek-beyond-react-16.html), at JSConf Iceland 2018.
+
+The application works like this: The more you type into the input below, the more detailed the charts below will get. Since both of the updates (the input element and the chart) run at the same time, the browser has to do so much computation that it will drop some frames. This leads to noticeable delays and a bad user experience:
+
+<video src="/blog/scheduling-in-react/sync-mode.mp4" muted="true" autoplay muted playsinline loop></video>
+
+However, in this example, it’s more important to update the input with the new text than it is to update the charts. A version that prioritizes the input will appear a lot more responsive to the end user even though the same computation time is required:
+
+<video src="/blog/scheduling-in-react/concurrent-mode.mp4" muted="true" autoplay muted playsinline loop></video>
+
+Unfortunately, current user interface architectures makes it non-trivial to implement this kind of prioritization. One way to solve this is by [debouncing](https://davidwalsh.name/javascript-debounce-function) the chart update. The problem with this is that the charts still render synchronously when the debounced callback fires, which will again cause our user interface to take multiple seconds during which it is not responsive. We can do better!
```

Comment:
Is it non-responsive for multiple seconds? Or does it take multiple seconds AND it's non-responsive during this time? Not sure if I'm asking clearly but basically wondering if this is "one" action or "two" actions?

"which will again cause our user interface to take multiple seconds during which it is not responsive."

Reply:
I just saw that I wrote "multiple seconds" which seems wrong we should make it **some time**.

I think it's one action that is the non-responsiveness for some time.

Reply:
What about

"which will again cause our user interface to be unresponsive for some time."

Or similar?

Reply:
Works for me. 🤗